src/realm/array_string.cpp

```diff
@@ -536,18 +536,39 @@ void ArrayString::verify() const
 #endif
 }
 
-ref_type ArrayString::write(_impl::ArrayWriterBase& out, StringInterner* interner)
+template <>
```

Comment:
Why a template?

Reply:
To avoid including the header file defining _impl::ArrayWriterBase in array_string.hpp